h2o-core/src/main/java/water/util/FrameUtils.java

```diff
@@ -334,25 +339,64 @@ public  double weightedMean() {
     final boolean _overwrite;
     final Job _j;
     int _nParts;
+    boolean _parallel;
     final CompressionFactory _compressor;
     final Frame.CSVStreamParams _csv_parms;
 
-    public ExportTaskDriver(Frame frame, String path, String frameName, boolean overwrite, Job j, int nParts,
+    public ExportTaskDriver(Frame frame, String path, String frameName, boolean overwrite, Job j, 
+                            int nParts, boolean perChunk,
                             CompressionFactory compressor, Frame.CSVStreamParams csvParms) {
       _frame = frame;
       _path = path;
       _frameName = frameName;
       _overwrite = overwrite;
       _j = j;
       _nParts = nParts;
+      _parallel = perChunk;
       _compressor = compressor;
       _csv_parms = csvParms;
     }
 
     @Override
     public void compute2() {
```

Comment:
Knit: So many can be final here - actually helps orientation in code.
Here is a patch to simply apply:
```patch
Index: h2o-core/src/main/java/water/util/FrameUtils.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/h2o-core/src/main/java/water/util/FrameUtils.java b/h2o-core/src/main/java/water/util/FrameUtils.java
--- a/h2o-core/src/main/java/water/util/FrameUtils.java	(revision 2fb82a2f9825d20bebae57b73f1705d35cce9ba1)
+++ b/h2o-core/src/main/java/water/util/FrameUtils.java	(date 1611654069770)
@@ -363,20 +363,20 @@
       if (_parallel && _nParts == 1) {
         _nParts = _frame.anyVec().nChunks();
         int processed = 0;
-        String compression = H2O.getSysProperty("export.csv.cache.compression", "none");
-        CompressionFactory compressor = CompressionFactory.make(compression);
-        DecompressionFactory decompressor = DecompressionFactory.make(compression);
-        String cacheStorage = H2O.getSysProperty("export.csv.cache.storage", "memory");
-        CsvChunkCache cache = "memory".equals(cacheStorage) ? new DkvCsvChunkCache() : new FileSystemCsvChunkCache();
+        final String compression = H2O.getSysProperty("export.csv.cache.compression", "none");
+        final CompressionFactory compressor = CompressionFactory.make(compression);
+        final DecompressionFactory decompressor = DecompressionFactory.make(compression);
+        final String cacheStorage = H2O.getSysProperty("export.csv.cache.storage", "memory");
+        final CsvChunkCache cache = "memory".equals(cacheStorage) ? new DkvCsvChunkCache() : new FileSystemCsvChunkCache();
         Log.info("Using compression=`" + compressor.getName() + 
                 "` and cache=`" + cache.getName() + "` for interim partial CSV export files.");
-        ChunkExportTask chunkExportTask = cache.makeExportTask(_frame, _csv_parms, compressor);
+        final ChunkExportTask chunkExportTask = cache.makeExportTask(_frame, _csv_parms, compressor);
         H2O.submitTask(new LocalMR(chunkExportTask, H2O.NUMCPUS));
         try (FileOutputStream os = new FileOutputStream(_path)) {
           byte[] buffer = new byte[BUFFER_SIZE];
-          boolean[] isChunkCompleted = new boolean[_nParts + 1];
+          final boolean[] isChunkCompleted = new boolean[_nParts + 1];
           while (processed != _nParts) {
-            int cid = chunkExportTask._completed.take();
+            final int cid = chunkExportTask._completed.take();
             isChunkCompleted[cid] = true;
             while (isChunkCompleted[processed]) {
               try (InputStream rawInputStream = cache.getChunkCsvStream(chunkExportTask, processed);
@@ -399,7 +399,7 @@
       } else if (_nParts == 1) {
         // Single file export, the file should be created by the node that was asked to export the data
         // (this is for non-distributed filesystems, we want the file to go to the local filesystem of the node)
-        Frame.CSVStream is = new Frame.CSVStream(_frame, _csv_parms);
+        final Frame.CSVStream is = new Frame.CSVStream(_frame, _csv_parms);
         exportCSVStream(is, _path, 0);
         tryComplete();
       } else {
@@ -408,7 +408,7 @@
           _nParts = calculateNParts(_csv_parms);
           assert _nParts > 0;
         }
-        int nChunksPerPart = ((_frame.anyVec().nChunks() - 1) / _nParts) + 1;
+        final int nChunksPerPart = ((_frame.anyVec().nChunks() - 1) / _nParts) + 1;
         new PartExportTask(this, _frame._names, nChunksPerPart, _csv_parms).dfork(_frame);
       }
     }
```